Src/String/String.cpp

```diff
@@ -433,74 +397,66 @@ bool String::equalsIgnoreCase(const String& other) const {
 }
 
 bool String::isEmpty() const {
-    return chs[0] == '\0';
+    return getChars()[0] == '\0';
 }
 
-void String::reallocate(int size, bool copyOldChs) {
+const String::CoreInfo String::reallocate(int size, bool copyOldChs) {
```

Comment:
This should still be void-returning

Reply:
Look at how this is used and its implementation, this is very much preferable.